src/vs/workbench/services/themes/browser/workbenchThemeService.ts

```diff
@@ -145,6 +144,8 @@ export class WorkbenchThemeService implements IWorkbenchThemeService {
 		// a color theme document with good defaults until the theme is loaded
 		let themeData: ColorThemeData | undefined = ColorThemeData.fromStorageData(this.storageService);
 		if (themeData && this.settings.colorTheme !== themeData.settingsId && this.settings.isDefaultColorTheme()) {
+			this.hasDefaultUpdated = themeData.settingsId === ThemeSettingDefaults.COLOR_THEME_DARK_OLD || themeData.settingsId === ThemeSettingDefaults.COLOR_THEME_LIGHT_OLD;
```

Comment:
I was expecting that this variable would be set to true if we were on the new themes, meaning that the settingsId would be equal to one of the new IDs. I'm wondering if I'm interpreting this line incorrectly?

Reply:
On shutdown, we store the current theme data (id and all colors) in the storage for fast recovery the next time vs code starts.
`themeData` is that persisted theme data.

It allows us to discover which user have been affected by the default change:
- if the user has a persisted `themeData`, then he's an existing user
- if 'themeData' points to an old default theme
- if the current theme is the default, but different from what's in the storage